src/finufft_core.cpp

```diff
@@ -1066,53 +1092,114 @@ int FINUFFT_PLAN_T<TF>::execute(std::complex<TF> *cj, std::complex<TF> *fk) {
     double t_pre = 0.0, t_spr = 0.0, t_t2 = 0.0,
            t_deconv = 0.0; // accumulated timings
     if (opts.debug)
-      printf("[%s t3] start ntrans=%d (%d batches, bsize=%d)...\n", __func__, ntrans,
-             nbatch, batchSize);
+      printf("[%s t3] start ntrans=%d (%d batches, bsize=%d)...\n", __func__,
+             ntrans_actual, nbatch, batchSize);
+
+    // allocate temporary buffers
+    // We are trying to be clever here and re-use memory whenever possible.
+    // Also, we allocate the memory for the "inner" NUFFT here as well,
+    // so that it doesn't need to be reallocated for every batch.
+    std::vector<TC, xsimd::aligned_allocator<TC, 64>> buf1, buf2, buf3;
```

Comment:
I guess saving this in the plan does not make sense.

Reply:
Then the memory savings would be lost.
As far as I can say from my own tests, the allocation is sufficiently cheap, but there may of course be counterexamples.

Reply:
small memory allocations, few pages is in the order of  < 10 ms

Reply:
Issues can arise, if
- memory is handed out directly by the kernel (e.g. via mmap),
- *and* several threads try to access the new pages simultaneously, creating "page fault races".

The kernel is usually quite good at bringing in new pages if there is no lock contention, but as soon as another thread wants to access a page while the kernel is loading this exact page, things slow down a lot.
However I don't think that is an issue here; I have encountered this only in some very specific circumstances so far.